src/sympc/protocol/fss/fss.py

```diff
@@ -232,99 +110,81 @@ def comp_evaluate(session: Session, b, x_masked, dtype=None) -> ShareTensor:
 
-        Returns:
-            TODO: Add return type
-        """
-        return dpf.keygen(n_values=n_values)
+    Currently supported operand are = & <=, respectively corresponding to
+    op = 'eq' and 'comp'.
 
-    @staticmethod
-    def eval(b, x, k_b):
-        """Sycret DPF eval.
+    Args:
+        x1 (MPCTensor): First private value.
+        x2 (MPCTensor): Second private value.
+        op: Type of operation to perform, should be 'eq' or 'comp'. Defaults to eq.
 
-        Args:
-            b: noqa: DAR101
-            x: noqa: DAR101
-            k_b: noqa: DAR101
+    Returns:
+        MPCTensor: Shares of the comparison.
+    """
+    assert not th.cuda.is_available()  # nosec
 
-        Returns:
-            noqa: DAR201
-        """
-        original_shape = x.shape
-        x = x.reshape(-1)
-        flat_result = dpf.eval(b, x, k_b)
-        return flat_result.astype(np.int32).astype(np.int64).reshape(original_shape)
+    # FIXME: Better handle the case where x1 or x2 is not a MPCTensor. For the moment
+    # FIXME: we cast it into a MPCTensor at the expense of extra communication
+    session = x1.session
+    dtype = session.tensor_type
 
+    shape = MPCTensor._get_shape("sub", x1.shape, x2.shape)
+    n_values = shape.numel()
 
-class DIF:
-    """Distributed Interval Function - used for comparison."""
+    CryptoPrimitiveProvider.generate_primitives(
+        f"fss_{op}",
+        sessions=session.session_ptrs,
+        g_kwargs={"n_values": n_values},
+        p_kwargs={},
+    )
 
-    @staticmethod
-    def keygen(n_values=1):
-        """Sycret DIF keygen.
+    args = zip(session.session_ptrs, x1.share_ptrs, x2.share_ptrs)
+    args = [list(el) + [op] for el in args]
 
-        Args:
-            n_values (int): Number of values. Defaults to 1
+    shares = parallel_execution(mask_builder, session.parties)(args)
 
-        Returns:
-            TODO: Add return type
-        """
-        return dif.keygen(n_values=n_values)
+    # TODO: don't do .reconstruct(), this should be done remotely between the evaluators
+    mask_value = MPCTensor(shares=shares, session=session)
+    mask_value = mask_value.reconstruct(decode=False) % 2 ** n
 
-    @staticmethod
-    def eval(b, x, k_b):
-        """Sycret DIF eval.
+    # TODO: add dtype to args
+    args = [
+        (session.session_ptrs[i], th.IntTensor([i]), mask_value, op) for i in range(2)
+    ]
 
-        Args:
-            b: noqa: DAR101
-            x: noqa: DAR101
-            k_b: noqa: DAR101
+    shares = parallel_execution(evaluate, session.parties)(args)
 
-        Returns:
-            # noqa: DAR201
-        """
-        # x = x.astype(np.uint64)
-        original_shape = x.shape
-        x = x.reshape(-1)
-        flat_result = dif.eval(b, x, k_b)
-        return flat_result.astype(np.int32).astype(np.int64).reshape(original_shape)
+    response = MPCTensor(session=session, shares=shares)
```

Comment:
NITPICK: You could add the `shape` in the `MPCTensor` (send it as an argument to the constructor)

Reply:
Done :)